src/main/kotlin/de/bund/digitalservice/useid/identification/IdentificationSessionsController.kt

```diff
@@ -35,40 +42,67 @@ class IdentificationSessionsController(
     }
 
     @PostMapping
-    fun createSession(@RequestBody createIdentitySessionRequest: CreateIdentitySessionRequest): Mono<ResponseEntity<CreateIdentitySessionResponse>> {
-        // Currently, returns a mock session response
-        return tcTokenUrlService.getTcTokenUrl()
-            .flatMap { tcTokenUrl ->
-                identificationSessionService.save(
-                    tcTokenUrl,
-                    createIdentitySessionRequest.refreshAddress,
-                    createIdentitySessionRequest.requestAttributes
-                )
-            }.map {
+    fun createSession(
+        @RequestBody createIdentitySessionRequest: CreateIdentitySessionRequest,
+        serverHttpRequest: ServerHttpRequest
+    ): Mono<ResponseEntity<CreateIdentitySessionResponse>> {
+        return identificationSessionService
+            .save(
+                refreshAddress = "https://localhost:8443/", // Currently a mock value, later will inject from env var mapped by API Key
+                requestAttributes = createIdentitySessionRequest.requestAttributes
+            ) // PublishOn is suggested by IntelliJ, maybe there is another way to resolve blocking URLEncoder.encode()
+            .publishOn(Schedulers.boundedElastic())
+            .map {
+                val encodedTcTokenUrl = URLEncoder.encode("${serverHttpRequest.uri}/$TCTOKEN_ENDPOINT/${it.useIDSessionId}", Utils.ENCODING)
                 ResponseEntity
                     .status(HttpStatus.OK)
                     .contentType(MediaType.APPLICATION_JSON)
-                    .body(
-                        CreateIdentitySessionResponse(
-                            tcTokenUrl = it.tcTokenUrl,
-                            sessionId = it.sessionId
-                        )
-                    )
-            }.doOnError { exception ->
-                log.error { "error occurred when creating identification session: ${exception.message}" }
-            }.onErrorReturn(
+                    .body(CreateIdentitySessionResponse(tcTokenUrl = encodedTcTokenUrl))
+            }
+            .doOnError { exception ->
+                log.error {
+                    "error occurred when creating identification session: ${exception.message}"
+                }
+            }
+            .onErrorReturn(
+                ResponseEntity.internalServerError().body(null)
+            )
+    }
+
+    @GetMapping("/$TCTOKEN_ENDPOINT/{useIDSessionId}")
+    fun getTCToken(@PathVariable useIDSessionId: UUID): Mono<ResponseEntity<TCTokenType>> {
+        return identificationSessionService.findByIdOrFail(useIDSessionId)
+            .flatMap {
+                tcTokenService.getTcToken(it.refreshAddress)
+            }
+            .doOnNext {
+                val eIDSessionId = UriComponentsBuilder
```

Comment:
Phew... 😁

Reply:
@mpanne is this the right way to get query param in Spring? or is there another way?

Reply:
That part looks fine for me, but it's a shame, that we can't get that ID from the token directly. I don't understand at the moment, why the sessionIdentifier is a different one, maybe we can find out it about it (later). It's just a weird way to get that id, even though I don't see a better one at the moment without researching about that behavior. Maybe you can write a research story for this, so that we can have a look at a later stage? Because it's working for now.

Reply:
@code28 also the expiration of the sessionId , which we don't know as well.

Reply:
I agree with Simon that we should find out more about the sessionIDs. But the way you extract the query parameter is fine.